shared/js/sk_key_utils.js

```diff
@@ -0,0 +1,367 @@
+(function (global) {
+  'use strict';
+
+  var softKeyModule = (function () {
+
+    // Keys which are mapped to accesskey
+    const SK_KEYS_MAPPING = {
+      LSK: 'a',
+      CSK: 's',
+      RSK: 'd'
+    };
+
+    // Storing the reference to softkeys
+    var sk_keys = {
+      LSK: null,
+      CSK: null,
+      RSK: null
+    };
+
+    // Container created for SoftKey
+    var sk_container = null;
+
+    // Check if menu open
+    var sk_options_menu = {
+      open: false,
+      cancelKey: SK_KEYS_MAPPING.LSK,
+      cancelKeyText: 'Cancel',
+      cancelKeyl10nId: 'cancel'
+    };
+
+    // Observer for observing the DOM changes
+    var observer = null;
+
+    return {
+      init: init
+    };
+
+    /**
+     * Used to initialize the library
+     * @example
+     * softKeyModule.init();
+     *
+     * @param N/A
+     * @return void 0
+     */
+    function init() {
+
+      // Calling destroy to destroy previously initialized variables
+      destroy();
+
+      // Selecting the softkey DOM element
+      var softKey = document.querySelector('[role="softkey"]');
+      if (!softKey) {
+        // If null, return
+        return;
+      }
+
+      // Container will be created to handle options
+      // menu and soft keys
+      createSoftKeyContainer();
+
+      for (var k in sk_keys) {
+        if (sk_keys.hasOwnProperty(k)) {
+
+          // Selecting the button using accesskey
+          sk_keys[k] =
+                  softKey.querySelector('[accesskey="' + SK_KEYS_MAPPING[k] + '"]');
+
+          if (sk_keys[k]) {
+
+            // Softkeys will be created for NON-Touch UI only
+            // A #IFDEF check is required
+            createSoftKeyButtons(sk_keys[k], k);
```

Comment:
And after creating a softkey for the element the other thing we need to do is hide the element. The element expects a click, but we don't have any way to click it, so we're replacing it with the soft key. This means that the unclickable element should not appear to the user.

It might be simplest, though, to just leave this to CSS and include styles that hide elements with accesskey attributes. That would give apps the ability to customize things. Some would be hidden and others would be display:none